src/Nox.Types/Types/Date/Date.cs

```diff
@@ -0,0 +1,149 @@
+    {
+        var newObject = new Date
+        {
+            Value = dateTime.Date,
+            _dateTypeOptions = dateTypeOptions
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Date"/> object.
+    /// </summary>
+    /// <param name="year">The year.</param>
+    /// <param name="month">The month.</param>
+    /// <param name="day">The day.</param>
+    /// <param name="dateTypeOptions">The date type options.</param>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Date From(int year, int month, int day, DateTypeOptions dateTypeOptions)
+    {
+        var dateTime = new DateTime(year, month, day);
+        return From(dateTime, dateTypeOptions);
+    }
+
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        if (Value.Date < _dateTypeOptions.MinValue)
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Number type as value {Value.Date.ToString("d", CultureInfo.InvariantCulture)} is less than than the minimum specified value of {_dateTypeOptions.MinValue.Date.ToString("d", CultureInfo.InvariantCulture)}"));
+        }
+
+        if (Value.Date > _dateTypeOptions.MaxValue)
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Number type as value {Value.Date.ToString("d", CultureInfo.InvariantCulture)} is greater than than the maximum specified value of {_dateTypeOptions.MaxValue.Date.ToString("d", CultureInfo.InvariantCulture)}"));
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Gets the year of the date.
+    /// </summary>
+    public int Year => Value.Year;
+
+    /// <summary>
+    /// Gets the month of the date.
+    /// </summary>
+    public int Month => Value.Month;
+
+    /// <summary>
+    /// Gets the day of the date.
+    /// </summary>
+    public int Day => Value.Day;
+
+    /// <summary>
+    /// Gets the day of week.
+    /// </summary>
+    public System.DayOfWeek DayOfWeek => Value.DayOfWeek;
+
+    /// <summary>
+    /// Gets the day of year.
+    /// </summary>
+    public int DayOfYear => Value.DayOfYear;
+
+    public override string ToString()
+        => ToString("d", CultureInfo.InvariantCulture);
+
+    public string ToString(string format)
+        => ToString(format, CultureInfo.InvariantCulture);
+
+    public string ToString(IFormatProvider formatProvider)
+        => ToString("d", formatProvider);
+
+    public string ToString(string format, IFormatProvider formatProvider)
+    {
+        if (!IsValidDateOnlyFormat(format))
+        {
+            throw new FormatException("Input string was not in a correct format.");
```

Comment:
This message is fine. I'm just wondering if maybe something like "The provided format is not supported." would be more descriptive?